azure/functions/decorators/function_app.py

```diff
@@ -2740,6 +2790,379 @@ def decorator():
+                         max_overlap: Optional[int] = 128,
+                         data_type: Optional[
+                             Union[DataType, str]] = None,
+                         **kwargs) \
+            -> Callable[..., Any]:
+        """
+        The embeddings input decorator creates embeddings which will be used to
+        measure the readiness of text strings.
+
+        Ref: https://platform.openai.com/docs/guides/embeddings
+
+        :param arg_name: The name of the variable that represents DaprState
+        output object in function code.
+        :param input: The input source containing the data to generate
+        embeddings for.
+        :param input_type: The type of the input.
+        :param model: The ID of the model to use.
+        :param max_chunk_length: The maximum number of characters to chunk the
+        input into. Default value: 8 * 1024
+        :param max_overlap: The maximum number of characters to overlap
+        between chunks. Default value: 128
+        :param data_type: Defines how Functions runtime should treat the
+        parameter value
+        :param kwargs: Keyword arguments for specifying additional binding
+        fields to include in the binding json
+
+        :return: Decorator function.
+        """
+
+        @self._configure_function_builder
+        def wrap(fb):
+            def decorator():
+                fb.add_binding(
+                    binding=EmbeddingsInput(
+                        name=arg_name,
+                        input=input,
+                        input_type=input_type,
+                        model=model,
+                        max_chunk_length=max_chunk_length,
+                        max_overlap=max_overlap,
+                        data_type=parse_singular_param_to_enum(data_type,
+                                                               DataType),
+                        **kwargs))
+                return fb
+
+            return decorator()
+
+        return wrap
+
+    def semantic_search_input(self,
+                              arg_name: str,
+                              connection_name: str,
+                              collection: str,
+                              query: Optional[str] = None,
+                              embeddings_model: Optional[OpenAIModels] = OpenAIModels.DefaultEmbeddingsModel,  # NoQA
+                              chat_model: Optional[OpenAIModels] = OpenAIModels.DefaultChatModel,  # NoQA
+                              system_prompt: Optional[str] = semantic_search_system_prompt,  # NoQA
+                              max_knowledge_count: Optional[int] = 1,
+                              data_type: Optional[
+                                  Union[DataType, str]] = None,
+                              **kwargs) \
+            -> Callable[..., Any]:
+        """
+        The embeddings input decorator creates embeddings which will be used to
+        measure the readiness of text strings.
```

Comment:
This one should be about Semantic Search

Reply:
Updated